text/0419-cdk-launchpads.md

```diff
@@ -0,0 +1,647 @@
+            "Properties": {
+              "PolicyDocument": {
+                "Statement": [
+                  {
+                    "Action": "iam",
+                    "Effect": "Deny",
+                    "Resource": "*"
+                  }
+                ],
+                "Version": "2012-10-17"
+              },
+              "Description": "The managed policy used as the permission boundary"
+            }
+          }
+        }
+      },
+      "export": {
+        "permissionBoundaryPolicy": {
+          "Ref": "PermissionBoundaryPolicy"
+        }
+      }
+    }
+  }
+}
+```
+
+And also with CDK, using the same CDK libraries that are used for applications:
+
+```ts
+import * as iam from 'aws-cdk-lib/aws-iam';
+
+const managedPolicy = new iam.ManagedPolicy(launchpad, 'Boundary', {
+  statements: [
+    new iam.PolicyStatement({
+      effect: iam.Effect.DENY,
+      actions: ['iam:'],
+      resources: [''],
+    }),
+  ],
+});
+// surface this resource to application developers using this Launchpad
+// (see below)
+launchpad.export(managedPolicy);
+```
+
+#### IAM Permission Boundaries
+
+Launchpads allow setting a given Managed Policy as the Permissions Boundary to be used by every application deployed into a given environment.
+
+You can do it through JSON/YAML:
+
+```json
+{
+  "launchpad": {
+    "iam": {
+      "permissionBoundary": {
+        "Ref": "PermissionBoundaryPolicy"
+      }
+    }
+  }
+}
+```
+
+Or in CDK code:
+
+```ts
+launchpad.setPermissionBoundaryPolicy(managedPolicy);
+```
+
+### Importing the Launchpad in CDK code
+
+Once a Launchpad has been deployed by the administrators,
+application developers can import it using the
+[`cdk-import` tool](https://github.com/cdklabs/cdk-import)
+(whose functionality will be incorporated into the main CDK CLI in the future).
+
+Executing the following command:
+
+```shell
+$ cdk-import launchpad -l typescript
+```
+
+Will generate code in your project that you then use in your CDK app definition:
+
+```ts
+import * as cdk from 'aws-cdk-lib';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import launchpad from '../build';
+
+const app = new cdk.App({
+    launchpad: launchpad, // adds the appropriate Permission Boundary automatically
+});
+const stack = new cdk.Stack(app, 'MyStack');
+const fn = new lambda.Function(stack, 'MyLambda', {
+  code: new lambda.Code.fromAsset('lambda-code'),
+  handler: 'index.handler',
+  runtime: lambda.Runtime.PYTHON_3_9,
+  // refer to a resource defined in the Launchpad
+  role: launchpad.Resources.functionRole,
+});
+```
+
+Additionally, importing the Launchpad in that way will produce a warning during `cdk deploy`
+if the version of the Launchpad in your environment is older than the current latest version of the Launchpad,
+and will ask you to update.
+
+### Integrating Launchpads with AWS Organizations
+
+Launchpads will integrate with AWS Organizations.
+When creating a new account using the AWS Console for Organizations,
```

Comment:
Similar to my other comment, would it be possible to use this with AWS Control Tower? Most of my clients now create new accounts via the AWS Control Tower "Landing Zone". "Landing Zone" and "Launchpad" even feels like a similar term 😄 

Since I'm assuming Control Tower is managed by another team(s), it's possible that it isn't feasible to integrate this feature with Control Tower. However, as an end-user, would be really great to integrate with an existing tool for simplicity.

Reply:
Yes, that's a good point. I'm not sure what exactly will that entail though, but this is an option we should explore internally.

Reply:
This AWS Solution might be interesting to consider as part of this: https://aws.amazon.com/blogs/architecture/fast-and-secure-account-governance-with-customizations-for-aws-control-tower/

Reply:
Lots of great ideas in this RFC, great reading -- As a customer I'd prefer to see Control Tower have an option to deploy the CDK bootstrap (no reason first person using CDK in the account should need to do that) as well as the LaunchPad config for the account.  It should happen both when onboarding existing accounts or when creating new accts (with Account Factory).  Pushing centrally managed rules out would be a very nice feature to have.